composer.json

```diff
@@ -20,7 +20,9 @@
   "type": "typo3-cms-extension",
   "version": "2.2.0",
   "require": {
-    "typo3/cms": ">=6.2.0 <7.9.99"
+    "php" : "^7.0.8",
+    "typo3/cms": "^7.6",
```

Comment:
In non-composer mode, I've just done a "composer update" inside the extension directory. The result is the download of a full typo3/cms in typo3conf/ext/dlf/vendor/ what is not nice at all. I removed the dependancy to typo3/cms which only loads solarium.

Actually, I have no good-practice, how to act with composer requirements in non-composer setups. :-(

Reply:
I'm not sure what I should do here now? Remove the typo3/cms requirement? Just leave it as it is?

Reply:
I think, there is no perfect solution to support both - composer mode and non-composer mode.

I would suggest to keep the requirement for typo3/cms, as it is the usual way, extension do. In composer mode, this is reasonable. I would drop requirement of PHP as this is declared by TYPO3 (7.6 requires PHP 7.0).

Finally I think this is a good choice:
<pre>
"require": {
    "typo3/cms": "^7.6",
    "solarium/solarium" : "^4.0"
  },
</pre>

In this case, we should add a note to the manual. In case of non-composer mode, the admin has to go to typo3conf/ext/dlf and run one "composer update". This will create the vendor directory and load solarium and unfortunately a bunch of things, we don't need here.

In ext_emconf.php we should not include the complete vendor-directory, as it contains TYPO3 itself. This could be confusing. The directory for the autoload should be "vendor/solarium".

Reply:
Do we need to add `vendor/solarium/` to the autoload section of composer.json as well?

Reply:
Please see my changes below: I added `vendor/` to the autoload classmap, but also excluded `vendor/typo3/` from autoloading.

Reply:
As far as I know TYPO3 7.6 requires at least PHP 5.5? Independently of that we can remove the PHP requirement, add vendor/solarium and vendor/symfony/event-dispatcher to autoload section of composer.json. Additionally I add a note to the manual for either run "composer update" once within the dlf directory or a manual download of the two dependencies without using composer. OK?

Reply:
Does it make sense to add empty directories vondor/solarium and vendor/symfony/event-dispatcher for an easier installation without composer?